docs/Relayer Fee Market.md

```diff
@@ -0,0 +1,51 @@
+4. Relayers and Users constitute a secondary supply-and-demand market, where prices rise when supply is low and fall when supply is abundant. There are no access restrictions for relayers, and anyone can enter. Relayer should evaluate and quote at their own discretion as an economically rational person. An incomplete list of risks that relayers should take into account is as follows:
+    1. Fluctuation in token prices and exchange ratios;
+    2. Time delay between quoting and claiming;
+    3. Loss of staking funds due to software or network failures
+
+## Proposal A-Three-tiered Quotation
+
+This approach is suitable for scenarios with lower gas fee on the source chain and shorter finality time. It has better versatility, reliability, and robustness. Such networks include Heco, BSC, Polygon and Darwinia.
+
+First, the relayer posts their quotes based on the reference price and the expected profit on the blockchain at any time. An off-chain pricing system maintains the reference price. Each relayer should lock a sufficient default margin on the chain to guarantee the faithful execution of the deal.
+
+In this way, a series of ***Ask*** prices come into being in the ascending order on the blockchain. When the user initiates a request on the source chain, the three lowest offers **_P1<P2<P3_** are filtered out and **P3**  is used as the billing price.  The user can request a cross-chain message delivery after paying **_P3_**. The reason that we select 3 relayers in the executed order is that we want to have redundancy for executing the message delivery. P1's relayer is R1, P2's relayer is R2, P3's relayer is R3.
+
+In any time, the message relayer and confirm relayer can be anyone, do not have be the assigned relayer. But in priority time slots, assigned relayer will be rewarded with much more regardless wo relay the message and do the confirmation.
+To give them different priority for **R1, R2, R3**, we will split the T into three consecutive time slots: **_0~T1 for P1_**,  **_T1~T2 for P2_**, and **_T2~T3 for P3_** respectively. The relayer who is assigned with her own time slot will be rewarded with more percentage, and this reward is for the asked price and the commitment(delivery in time or slash). Relayer with lower price are assigned with earlier time slot.
+### Detailed Steps in Implementation
+
+1. Enroll and lock collateral
+    1. `enroll_and_lock_collateral` dispatch call
+    2. `cancel_enrollment()` dispatch call, remember to check if the relayer is in priority time slots.
+2. Ask price
```

Comment:
How about changing `price` to  `fee`, Since this is Fee market design doc, not Price market doc.

Ask price => Ask fee
P1 => F1

Reply:
Price here indicate `fee per message`.

`Fee = ∑(price_fn(message))`

In crypto economics, the convention is to call it fee market, but in this specific step, I think we are good to use price, can be seen as abbreviation of `fee per message`.